CoffeeMachineSimulator/CoffeeMachineSimulator.Services/Services/CoffeeService.cs

```diff
@@ -19,13 +19,19 @@ public CoffeeService()
         public void AddCoffee(CoffeeModel coffeeToAdd)
         {
             if (coffeeToAdd == null) throw new Exception("You should not add null entries!");
-
-            coffeeModels.Add(coffeeToAdd);
+            
+            if (coffeeModels.Any(i => i == coffeeToAdd)) throw new Exception("Already exists an entry with the same Id");
+            if (!(coffeeToAdd.Id == Guid.Empty || coffeeToAdd.Name == null || coffeeToAdd.Price <= 0.0f || coffeeModels.Any(j => j.Name == coffeeToAdd.Name) || coffeeModels.Any(j => j.Id == coffeeToAdd.Id) || coffeeModels.Any(j => j.Price == coffeeToAdd.Price)))
```

Comment:
please make a method that does all of these checks. No line of code should be that big in order to scroll to see.